### PR TITLE
modify ValidateHierarchy interface

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#352390](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/352390)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6938)

### CODE REVIEW

**Comment by xichan-ms** [comment]
@fkpwolf @mingwli pls help to review.

**Comment by xichan-ms** [comment]
Diff is no related to this pr.

File: src/Microsoft.Docs.LearnValidation/Validators/ModuleValidator.cs [hunk 1]
```diff
@@ -30,13 +30,13 @@ public override bool Validate(Dictionary<string, IValidateModel> fullItemsDict)
                 }
 
                 // module has child unit, but that unit has error when SDP validating
-                var childrenCantFind = module.Units.Where(u => !fullItemsDict.ContainsKey(u));
-                if (childrenCantFind.Any())
+                var childrenCantFind = module?.Units.Where(u => !fullItemsDict.ContainsKey(u));
```
Comment (herohua) [inline src/Microsoft.Docs.LearnValidation/Validators/ModuleValidator.cs:33]: line 27 already assumes `module != null` - should we update that line as well?

Comment (herohua) [reply src/Microsoft.Docs.LearnValidation/Validators/ModuleValidator.cs:33]: how about `continue` if `module == null`

Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/Validators/ModuleValidator.cs:33]: Thanks, will fix it.

Comment (xichan-ms) [reply src/Microsoft.Docs.LearnValidation/Validators/ModuleValidator.cs]: Sorry, I mixed up two PR, the issue possible null reference is tracked by another existing [PR](https://github.com/dotnet/docfx/pull/6918).

**Review by yufeih — APPROVED** [review]

**Comment by xichan-ms** [comment]
Diff expected.

**Pull request merged**